sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -52,38 +53,66 @@ def _parse_conn_str(conn_str, kwargs):
-        if key.lower() == "endpoint":
-            endpoint = value.rstrip("/")
-        elif key.lower() == "hostname":
-            endpoint = value.rstrip("/")
-        elif key.lower() == "sharedaccesskeyname":
-            shared_access_key_name = value
-        elif key.lower() == "sharedaccesskey":
-            shared_access_key = value
-        elif key.lower() == "entitypath":
-            entity_path = value
-        elif key.lower() == "sharedaccesssignature":
-            shared_access_signature = value
-            try:
-                # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
-                # type: ignore
-                shared_access_signature_expiry = int(shared_access_signature.split('se=')[1].split('&')[0])
-            except (IndexError, TypeError, ValueError): # Fallback since technically expiry is optional.
-                # An arbitrary, absurdly large number, since you can't renew.
-                shared_access_signature_expiry = int(time.time() * 2)
-    if not (all((endpoint, shared_access_key_name, shared_access_key)) or all((endpoint, shared_access_signature))):
+    check_case = kwargs.pop("check_case", False)    # type: Optional[bool]
+    conn_settings = core_parse_connection_string(conn_str, case_sensitive_keys=check_case)
+    if check_case:
+        shared_access_key = conn_settings.get("SharedAccessKey")
+        shared_access_key_name = conn_settings.get("SharedAccessKeyName")
+        endpoint = conn_settings.get("Endpoint")
+        entity_path = conn_settings.get("EntityPath")
+    
+        # non case sensitive check when parsing connection string for internal use
+        for key, value in conn_settings.items():
+            # only sas check is non case sensitive for both conn str properties and internal use
+            if key.lower() == "sharedaccesssignature":
+                shared_access_signature = value
+                try:
+                    # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
+                    shared_access_signature_expiry = int(
+                        shared_access_signature.split("se=")[1].split("&")[0]   # type: ignore
+                    )
+                except (
+                    IndexError,
+                    TypeError,
+                    ValueError,
+                ):  # Fallback since technically expiry is optional.
+                    # An arbitrary, absurdly large number, since you can't renew.
+                    shared_access_signature_expiry = int(time.time() * 2)
+
+    if not check_case:
+        endpoint = conn_settings.get("endpoint") or conn_settings.get("hostname")
+        if endpoint:
+            endpoint = endpoint.rstrip("/")
+        shared_access_key_name = conn_settings.get("sharedaccesskeyname")
+        shared_access_key = conn_settings.get("sharedaccesskey")
+        entity_path = conn_settings.get('entitypath')
```

Comment:
nit: ' -> " for consistency
```suggestion
        entity_path = conn_settings.get("entitypath")
```